quiche/src/recovery/mod.rs

```diff
@@ -572,7 +572,16 @@ impl ReleaseDecision {
 /// Recovery statistics
 #[derive(Default, Debug)]
 pub struct RecoveryStats {
-    pub startup_exit: Option<StartupExit>,
+    startup_exit: Option<StartupExit>,
+}
+
+impl RecoveryStats {
+    // Record statistics when a CCA first exits startup.
+    pub fn set_startup_exit(&mut self, startup_exit: StartupExit) {
+        if let None = self.startup_exit {
```

Comment:
Possible nit, this is equivalent to: `if self.startup_exit.is_none() {`

Clippy seems to have a similar opinion:
```
error: redundant pattern matching, consider using `is_none()`
   --> quiche/src/recovery/mod.rs:581:16
    |
581 |         if let None = self.startup_exit {
    |         -------^^^^-------------------- help: try: `if self.startup_exit.is_none()`
    |
    = help: for further information visit [https://rust-lang.github.io/rust-](https://rust-lang.github.io/rust-clippy/master/index.html#redundant_pattern_matching)
```